features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/EndpointsNew/GenericEndpoint.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c)  WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { Divider, Icon, IconButton, InputAdornment, TextField, withStyles } from '@material-ui/core';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+const styles = theme => ({
+    endpointInputWrapper: {
+        width: '100%',
+        display: 'flex',
+        justifyContent: 'space-between',
+    },
+    textField: {
+        width: '100%',
+    },
+    input: {
+        marginLeft: theme.spacing.unit,
+        flex: 1,
+    },
+    iconButton: {
+        padding: theme.spacing.unit,
+    },
+    divider: {
+        width: 1,
+        height: 28,
+        margin: 4,
+    },
+});
+function GenericEndpoint(props) {
+    const {
+        category,
+        endpointURL,
+        editEndpoint,
+        classes,
+        type,
+        setAdvancedConfigOpen,
+        deleteEndpoint,
+        index,
+    } = props;
+    const [serviceUrl, setServiceUrl] = useState(endpointURL);
+
+    useEffect(() => {
+        setServiceUrl(endpointURL);
+    }, [props]);
```

Comment:
isn't it better if we bind the precondition to set setServiceUrl, for a specific prop passed rather than using the whole props object?

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/6725/commits/625591235cad25fe1786ea5547d1f85d34a75dd2